kong/plugins/bot-detection/cache.lua

```diff
@@ -0,0 +1,27 @@
+local cache = require "kong.tools.database_cache"
+
+local _M = {}
+
+local INDEX = "bot_detection_index"
+
+function _M.set(key, value)
+  cache.set(cache.bot_detection_key(key), value)
+  local index_keys = cache.get(INDEX)
+  if not index_keys then index_keys = {} end
+  table.insert(index_keys, key)
```

Comment:
`index_keys[#index_keys+1] = key` is the preferred way because faster and cleaner.